### PR TITLE
[Design] [MIS] Employee Management Module

### LINKED ISSUE #366: [Design] [MIS] MIS - Employee Management
## Content

- MIS

## Rubric Sections

- DetDesDoc: Syntax
- DetDesDoc: Semantics
- DetDesDoc: EnoughToBuild
- Important Parts Formalized (LO_SpecMath)

### PR DESCRIPTION
**Pull request by joyxiao99**
close https://github.com/arkinmodi/project-sayyara/issues/366

### CODE REVIEW

File: docs/Design/SoftDetailedDes/MIS.tex [hunk 1]
```diff
@@ -812,11 +813,104 @@ \subsubsection{Access Routine Semantics}
 
 \subsubsection{Local Functions}
 
-\noindent hash($s$):
+\section{MIS of Employee Management Module} \label{mEmployeeManagement}
```
Comment (arkinmodi) [inline docs/Design/SoftDetailedDes/MIS.tex:872]:
```suggestion
\newpage

\section{MIS of Employee Management Module} \label{mEmployeeManagement}
```

File: docs/Design/SoftDetailedDes/MIS.tex [hunk 2]
```diff
@@ -812,11 +813,104 @@ \subsubsection{Access Routine Semantics}
...
+\subsubsection{Exported Access Programs}
+
+\begin{center}
+	\begin{adjustbox}{width=\textwidth}
+		\begin{tabular}{llll}
+			\hline
+			\textbf{Name}   & \textbf{In}                & \textbf{Out}          & \textbf{Exceptions} \\
+			\hline
+			getEmployeeById & String                     & $Employee$            & ~                   \\
+			getAllEmployees & String                     & List<$Employee$>      & ~                   \\
```
Comment (arkinmodi) [inline docs/Design/SoftDetailedDes/MIS.tex]:
```suggestion
			getAllEmployees & String                     & List$<$Employee$>$    & ~                   \\
```

File: docs/Design/SoftDetailedDes/MIS.tex [hunk 3]
```diff
@@ -812,11 +813,104 @@ \subsubsection{Access Routine Semantics}
...
+			updateEmployee  & String, UpdateEmployeeType & UserNotFoundException                       \\
```
Comment (arkinmodi) [inline docs/Design/SoftDetailedDes/MIS.tex]:
```suggestion
updateEmployee  & String, UpdateEmployeeType & ~                  & UserNotFoundException \\
```

File: docs/Design/SoftDetailedDes/MIS.tex [hunk 4]
```diff
@@ -812,11 +813,104 @@ \subsubsection{Access Routine Semantics}
...
 \begin{itemize}
-	\item output: $out := \text{hashed value of } s$
+	\item output: $out :=$ An employee such that it's ID matches $employeeId$ from the employee database
+	      table else None.
 \end{itemize}
 
+\noindent getAllEmployees($shopId$):
+\begin{itemize}
+	\item output: $out := List of employees from the shop database table associated with the $shopId$, else
+		      None.$
```
Comment (arkinmodi) [inline docs/Design/SoftDetailedDes/MIS.tex]:
```suggestion
	\item output: $out :=$ List of employees from the shop database table associated with the $shopId$, else
	      None.
```

File: docs/Design/SoftDetailedDes/MIS.tex [hunk 5]
```diff
@@ -812,11 +813,104 @@ \subsubsection{Access Routine Semantics}
...
+	\item output: $out := getEmployeeById(id) = \text{Employee} \Rightarrow (\text{Employee} = \text{None} $
+		      \\ $\Rightarrow \text{UserNotFoundException } |\ \text{Employee} \neq \text{None}$ \\ $\Rightarrow
+		      \text{update all fields of Employee with fields in } patch \text{ in service database table})$
+	\item exception: $exc := getEmployeeById(id) = None \Rightarrow \text{UserNotFoundException}$
+\end{itemize}
+
+\noindent deleteEmployee($employeeId$):
+\begin{itemize}
+	\item transition: Delete employee where their ID matches $employeeId$ from the employee database table.
+\end{itemize}
```
Comment (arkinmodi) [inline docs/Design/SoftDetailedDes/MIS.tex]: Based on what Leon said yesterday, deleting an employee might not be what we want to do. 
cc: @LeonSo7

Screenshot 2023-01-16 at 4 48 07 PM

File: docs/Design/SoftDetailedDes/MIS.tex [hunk 6]
```diff
@@ -812,11 +813,104 @@ \subsubsection{Access Routine Semantics}
...
+
+\subsubsection{Exported Access Programs}
+
+\begin{center}
+	\begin{adjustbox}{width=\textwidth}
+		\begin{tabular}{llll}
+			\hline
+			\textbf{Name}   & \textbf{In}                & \textbf{Out}          & \textbf{Exceptions} \\
+			\hline
+			getEmployeeById & String                     & $Employee$            & ~                   \\
```
Comment (arkinmodi) [inline docs/Design/SoftDetailedDes/MIS.tex]:
```suggestion
			getEmployeeById & String                     & Employee              & ~                   \\
```

**Review by arkinmodi — CHANGES_REQUESTED** [review]

File: docs/Design/SoftDetailedDes/MIS.tex [hunk 7]
```diff
@@ -864,11 +867,107 @@ \subsubsection{Access Routine Semantics}
...
+\noindent updateEmployee($employeeId, patch$):
+\begin{itemize}
+	\item transition: Update all fields of Employee with fields in $patch$ in employee database table
+	\item output: $out := getEmployeeById(id) = \text{Employee} \Rightarrow (\text{Employee} = \text{None} $
+		      \\ $\Rightarrow \text{UserNotFoundException } |\ \text{Employee} \neq \text{None}$ \\ $\Rightarrow
+		      \text{update all fields of Employee with fields in } patch \text{ in service database table})$
+	\item exception: $exc := getEmployeeById(id) = None \Rightarrow \text{UserNotFoundException}$
+\end{itemize}
+
+\noindent removeEmployee($employeeId$):
```
Comment (LeonSo7) [inline docs/Design/SoftDetailedDes/MIS.tex]:
```suggestion
\noindent suspendEmployee($employeeId$):
```

File: docs/Design/SoftDetailedDes/MIS.tex [hunk 8]
```diff
@@ -864,11 +867,107 @@ \subsubsection{Access Routine Semantics}
...
+\begin{center}
+	\begin{adjustbox}{width=\textwidth}
+		\begin{tabular}{llll}
+			\hline
+			\textbf{Name}   & \textbf{In}                & \textbf{Out}       & \textbf{Exceptions}   \\
+			\hline
+			getEmployeeById & String                     & Employee           & ~                     \\
+			getAllEmployees & String                     & List$<$Employee$>$ & ~                     \\
+			updateEmployee  & String, UpdateEmployeeType & ~                  & UserNotFoundException \\
+			removeEmployee  & String                     & ~                  & ~                     \\
```
Comment (LeonSo7) [inline docs/Design/SoftDetailedDes/MIS.tex]:
```suggestion
			suspendEmployee  & String                     & ~                  & ~                     \\
```

File: docs/Design/SoftDetailedDes/MIS.tex [hunk 9]
```diff
@@ -864,11 +867,107 @@ \subsubsection{Access Routine Semantics}
...
+\subsubsection{Exported Constants}
+
+None
+
+\subsubsection{Exported Types}
+
+\textbf{UpdateEmployeeType}
+
+\begin{table}[H]
+	\begin{tabular}{|l|l|l|}
```
Comment (arkinmodi) [inline docs/Design/SoftDetailedDes/MIS.tex]: I'm updating these in my MIS for Quotes PR.

```suggestion
	\begin{tabular}{|p{0.25\textwidth}|p{0.25\textwidth}|p{0.40\textwidth}|}
```

**Review by arkinmodi — APPROVED** [review]

**Pull request merged**